tests/bluetooth/tester/src/audio/btp_pbp.c

```diff
@@ -0,0 +1,267 @@
+
+	tester_event(BTP_SERVICE_ID_PBP, BTP_PBP_EV_PUBLIC_BROADCAST_ANOUNCEMENT_FOUND, ev_ptr,
+		     sizeof(*ev_ptr) + broadcast_name_len);
+
+	tester_rsp_buffer_free();
+	tester_rsp_buffer_unlock();
+}
+
+static struct bt_le_scan_cb pbp_scan_cb = {
+	.recv = pbp_scan_recv,
+};
+
+static uint8_t pbp_read_supported_commands(const void *cmd, uint16_t cmd_len, void *rsp,
+					   uint16_t *rsp_len)
+{
+	struct btp_pbp_read_supported_commands_rp *rp = rsp;
+
+	tester_set_bit(rp->data, BTP_PBP_READ_SUPPORTED_COMMANDS);
+	tester_set_bit(rp->data, BTP_PBP_SET_PUBLIC_BROADCAST_ANNOUNCEMENT);
+	tester_set_bit(rp->data, BTP_PBP_SET_BROADCAST_NAME);
+	tester_set_bit(rp->data, BTP_PBP_BROADCAST_SCAN_START);
+	tester_set_bit(rp->data, BTP_PBP_BROADCAST_SCAN_STOP);
+
+	*rsp_len = sizeof(*rp) + 1;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static int pbp_broadcast_source_adv_setup(void)
+{
+	struct bt_le_adv_param param =
+		BT_LE_ADV_PARAM_INIT(0, BT_GAP_ADV_FAST_INT_MIN_2, BT_GAP_ADV_FAST_INT_MAX_2, NULL);
+	uint32_t gap_settings = BIT(BTP_GAP_SETTINGS_DISCOVERABLE) |
+				BIT(BTP_GAP_SETTINGS_EXTENDED_ADVERTISING);
+	uint32_t broadcast_id;
+
+	NET_BUF_SIMPLE_DEFINE(ad_buf, BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE);
+	NET_BUF_SIMPLE_DEFINE(pba_buf, BT_UUID_SIZE_16 + 3 + pbp_metadata_cached_len);
+
+	int err = bt_rand(&broadcast_id, BT_AUDIO_BROADCAST_ID_SIZE);
+	struct bt_data ext_ad[3];
+
+	if (err) {
+		LOG_ERR("Unable to generate broadcast ID: %d\n", err);
+		return -EINVAL;
+	}
+
+	ext_ad[0].type = BT_DATA_BROADCAST_NAME;
+	ext_ad[0].data_len = pbp_name_cached_len;
+	ext_ad[0].data = pbp_broadcast_name_cached;
+	net_buf_simple_add_le16(&ad_buf, BT_UUID_BROADCAST_AUDIO_VAL);
+	net_buf_simple_add_le24(&ad_buf, broadcast_id);
+	ext_ad[1].type = BT_DATA_SVC_DATA16;
+	ext_ad[1].data_len = ad_buf.len;
+	ext_ad[1].data = ad_buf.data;
+	net_buf_simple_add_le16(&pba_buf, BT_UUID_PBA_VAL);
+	net_buf_simple_add_u8(&pba_buf, pbp_features_cached);
+	net_buf_simple_add_u8(&pba_buf, pbp_metadata_cached_len);
+	net_buf_simple_add_mem(&pba_buf, pbp_metadata_cached, pbp_metadata_cached_len);
+	ext_ad[2].type = BT_DATA_SVC_DATA16;
+	ext_ad[2].data_len = pba_buf.len;
+	ext_ad[2].data = pba_buf.data;
+
+	err = tester_gap_create_adv_instance(&param, BTP_GAP_ADDR_TYPE_IDENTITY, ext_ad,
+					     ARRAY_SIZE(ext_ad), NULL, 0, &gap_settings);
+	if (err) {
+		LOG_ERR("Could not set up extended advertisement: %d", err);
+		return -EINVAL;
+	}
+
+	cap_extern_ext_ad_setup();
+
+	return 0;
+}
+
+static uint8_t pbp_set_public_broadcast_announcement(const void *cmd, uint16_t cmd_len,
+						     void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_pbp_set_public_broadcast_announcement_cmd *cp = cmd;
+	int err = -EINVAL;
+
+	if (cp->metadata_len <= PBP_EXT_ADV_METADATA_LEN_MAX) {
+		pbp_features_cached = cp->features;
+		pbp_metadata_cached_len = cp->metadata_len;
+		memcpy(pbp_metadata_cached, cp->metadata, cp->metadata_len);
+		err = pbp_broadcast_source_adv_setup();
```

Comment:
Hmm, both `pbp_set_public_broadcast_announcement` and `pbp_set_broadcast_name` creates the extended advertising set. 

Shouldn't `pbp_set_broadcast_name` only update the advertising data?

Reply:
This is because we need both to create a set, and I take no assumption on which is set first. The second set created replaces the incomplete one, regardless of in which order pbp_set_public_broadcast_announcement and pbp_set_broadcast_name are called.

Reply:
Note that pbp_broadcast_source_adv_setup does not actually start the advertisement. It just sets up an advertisement instance. Actually starting it is done later during the call to cap_broadcast_source_setup.